docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,555 @@
+}
+```
+
+</details>
+
+### Step 3: Deploy the Extension Contract
+
+We will use the abi and bytecode to deploy the extension contract
+
+```js
+import { abi as TipMeABI } from './TipMe.json';
+import { bytecode as TipMeBytecode } from './TipMe.json';
+
+const TipMeFactory = new ethers.ContractFactory(
+  TipMeABI,
+  TipMeBytecode,
+  signer,
+);
+
+const tipMeContract = await TipMeFactory.deploy();
+await tipMeContract.deployTransaction.wait();
+
+const tipMeExtensionAddress = tipMeContract.addres;
+```
+
+### Step 4: Encode Function Selector and Store in Universal Profile
+
+:::info Requirements
+
+The address calling the `setData(..)` function needs to have `ADDEXTENSIONS` and `CHANGEEXTENSIONS` permission otherwise the call will fail. Check the [keyManager guides](../key-manager/get-controller-permissions.md) to learn more about permissions.
+
+:::
+
+We need to encode the function selector of `tipMe(..)` and store it in the Universal Profile with a specific data key according to the [LSP17-ContractExtension](../../../standards/generic-standards/lsp17-contract-extension.md) standard.
+
+#### Prepare data key and data value
+
+```js
+import { ERC725YDataKeys } from '@lukso/lsp-smart-contracts';
+
+// We get `TipMeABI` from the last step
+// import {abi as TipMeABI} from './TipMe.json';
+const tipMeInterface = new ethers.utils.Interface(TipMeABI);
+const tipMeFunctionSelector = tipMeInterface.getSighash('tipMe()');
+
+// Define the DataKey for the extension
+// according to the LSP17-ContractExtension standard
+const extensionDataKey =
+  ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
+  tipMeFunctionSelector.substr(2) +
+  '0'.repeat(32);
+```
+
+#### Controlling Value Forwarding to Extensions
+
+When calling an extension through a UniversalProfile, **in case value was sent along the call**, there's an option to either retain the sent value within the UniversalProfile contract or forward it to the extension contract. This choice depends on the intended purpose of the extension. In scenarios where a single extension is utilized collectively by multiple users, it might be preferable to keep the value within the UniversalProfile. Alternatively, if the extension requires funds to perform specific actions, forwarding the value might be necessary.
```

Comment:
```suggestion
When calling an extension through a UniversalProfile, **in case value was sent along the call**, there's an option to either retain the sent value within the UniversalProfile contract or forward it to the extension contract. This choice depends on the intended purpose of the extension. In scenarios where multiple users collectively use a single extension, keeping the value within the UniversalProfile might be preferable. Alternatively, forwarding the value might be necessary if the extension requires funds to perform specific actions.
```